dbms/src/Core/AccurateComparison.h

```diff
@@ -41,8 +41,8 @@ using bool_if_not_safe_conversion = std::enable_if_t<!is_safe_conversion<A, B>::
 /// Case 2. Are params IntXX and UIntYY ?
 template <typename TInt, typename TUInt>
 using is_any_int_vs_uint = std::integral_constant<bool,
-                            std::is_integral<TInt>::value && std::is_integral<TUInt>::value &&
-                               std::is_signed<TInt>::value && std::is_unsigned<TUInt>::value>;
+                            std::is_integral<TInt>::value && (std::is_integral<TUInt>::value || std::is_same<TUInt, DB::UInt128>::value) &&
+                               std::is_signed<TInt>::value && (std::is_unsigned<TUInt>::value || std::is_same<TUInt, DB::UInt128>::value)>;
```

Comment:
Specializing std::is_integral was my first idea. But the problem it's in some part of the code they are using std::is_integral.

For example, in `DataTypes/DataTypeNumberBase.cpp`, we have the `readTextUnsafeIfIntegral` function member who call the `readIntTextUnsafe` function member (`IO/ReadHelpers.h`).

The problem now it's this function call fews operators who are not overloaded like `operator*=` or `operator+=` which is a little bit complicated because the `UInt128` only emulate fews functions that a natural type have (principally comparison operator) but the arithmetic operator aren't overloaded because it is not really an unsigned integral but only a simulation of it.

A solution I can suggest it's maybe to overload `std::is_integral` (with value as true) and `std::is_arithmetic`(as false) and check in fews functions who need the arithmetic operator not only `std::is_integral` but also `std::is_arithmetic`, but maybe it's a little bit overkill for a single type ?